### PR TITLE
feat: Enable/Disable downloads for different roles

### PR DESCRIPTION
**Pull request by gitstart**
## Change Summary
ref: #4879

There is an option for shared views to disable download for anonymous users which is cool.
But if a user is logged in as a collaborator he/she can download all tables data.

## Change type
- [x]  feat: (new feature for the user, not a new feature for build script)

## Additional information / screenshots (optional)
https://www.loom.com/share/53fbafc6c21c431094d63a29a66d5560

---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/toolbar/ViewActions.vue [hunk 1]
```diff
@@ -138,7 +140,7 @@ useMenuCloseOnEsc(open)
 
             <a-menu-divider />
 
-            <a-sub-menu key="download">
+            <a-sub-menu v-if="hasRole('owner', true)" key="download">
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/toolbar/ViewActions.vue:143]: By doing so, all editors wouldn't be able to download as CSV / XLSX. You should make it configurable to cover both use cases.

Comment (gitstart) [reply packages/nc-gui/components/smartsheet/toolbar/ViewActions.vue:143]: thank you for your review @wingkwong , will look into it

**Comment by o1lab** [comment]
@gitstart : as one of our engineer, wing, has mentioned - please comment on the issue that will be worked on so that we can discuss the solution first. At times, what looks like an easy solution will have a lot more beneath. It will help optimise your and our efforts.

**Comment by frankMagoba** [comment]
> @gitstart : as one of our engineer, wing, has mentioned - please comment on the issue that will be worked on so that we can discuss the solution first. At times, what looks like an easy solution will have a lot more beneath. It will help optimise your and our efforts.

Hi @o1lab , this is well noted, thanks. Need a quick clarrification, can we use the same UI for sharing views with anonymous users, and such that when "Download allowed" is unchecked, other collaborators can't download the tables?

Also are the events for disabling these actions on anonymous views also used in collaborators view? Thanks.

**Comment by wingkwong** [comment]
Closing - inactivity.

**Pull request closed without merging**